tapchannel/aux_sweeper.go

```diff
@@ -322,53 +397,98 @@ func (a *AuxSweeper) resolveContract(req lnwallet.ResolutionReq,
 	//  * complete all the proofs w/ the added txn
 	//  * CreateProofSuffixCustom?
 
-	//contractPoint := req.ContractPoint
+	var (
+		sweepDesc    lfn.Result[tapscriptSweepDesc]
+		assetOutputs []*cmsg.AssetOutput
+	)
 
 	switch req.Type {
 	// A non-delay output. This means the remote party force closed on
 	// chain.
-	case lnwallet.CommitNoDelay:
+	case input.TaprootRemoteCommitSpend:
```

Comment:
Nice, this whole `switch` statement makes it very obvious and easy to understand what's going on :100: Cool that you were able to generalize the code so much.

Reply:
Yee, took a few iterations to get it into this current version. The logic is focused, as it just worries about how to make a sweep desc for a specific asset type, then the rest _should_ be able to stay the same. 

Note that it isn't aware yet of HTLC preimages, but my plan is to add that to the `tapscriptSweepDesc` struct. 

Initially I was going to refactor the lnd witness generation functions a bit more to re-use, but they're all mostly the same, and only the HTLC ones really deviate much. So opted to use as inspiration and lift what was identical.